CalcSolverSsr/src/components/BodyMassIndex.tsx

```diff
@@ -0,0 +1,95 @@
+import React, {useEffect, useState} from "react";
+
+const BodyMassIndex = () => {
+    useState()
+
+    const [weight, setWeight] = useState(0)
+    const [height, setHeight] = useState(0)
+    const [bmi, setBmi] = useState(0)
+    const [heightUnit, setHeightUnit] = useState("inches")
+    const [massUnit, setMassUnit] = useState("lbs")
+
+    useEffect(() => {
+        updateBmi()
+    })
+
+    const roundNumber = (num: number, dec: number) => {
+        return Math.round(num * Math.pow(10, dec)) / Math.pow(10, dec);
+    }
+
+    const updateBmi = () => {
+        let newWeight;
+        if (massUnit !== 'kg') {
+            newWeight = convertPoundsToKilograms(weight)
+
+        } else {
+            newWeight = weight
+        }
+        let newHeight
+        if (heightUnit !== 'cm') {
+            newHeight = convertInchesToCentimeters(height)
+        } else {
+            newHeight = height
+        }
+
+
+        setBmi(newWeight / Math.pow(newHeight / 100, 2))
+    }
+
+    const convertPoundsToKilograms = (valueInPounds: number) => {
+        return 0.45359237 * valueInPounds;
+    }
+
+    const convertInchesToCentimeters = (valueInInches: number) => {
+        return 2.54 * valueInInches;
+    }
+
+    return (
+        <div>
+            <h1>BMI Calculator</h1>
+            <div className="input-group">
+                <div className="input-group-prepend">
+                    <span className="input-group-text">Weight</span>
+                </div>
+                <input className="form-control"
+                       type="number"
```

Comment:
Try to condense the multiple properties into a single line. Makes it more consistent and easier to read.